src/Functions/FunctionsCodingUUID.cpp

```diff
@@ -13,43 +13,159 @@
 #include <Interpreters/Context_fwd.h>
 #include <Interpreters/castColumn.h>
 
-namespace DB
+#include <span>
+
+namespace DB::ErrorCodes
+{
+extern const int ARGUMENT_OUT_OF_BOUND;
+extern const int ILLEGAL_COLUMN;
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+extern const int NOT_IMPLEMENTED;
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+enum class Representation
+{
+    BigEndian,
+    LittleEndian
+};
+
+std::pair<int, int> determineBinaryStartIndexWithIncrement(const ptrdiff_t num_bytes, const Representation representation)
+{
+    if (representation == Representation::BigEndian)
+        return {0, 1};
+    else if (representation == Representation::LittleEndian)
+        return {num_bytes - 1, -1};
+
+    throw DB::Exception(DB::ErrorCodes::NOT_IMPLEMENTED, "{} is not implemented yet", magic_enum::enum_name(representation));
```

Comment:
probably `LOGICAL_ERROR` would fit better. we cannot get here without a bug in code, right?

Reply:
Sure, I agree.